src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,267 @@
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NIndex> startIndex);
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> range);
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<nint> start);
+        void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        void FlattenTo(System.Span<T> destination);
+        void GetLengths(System.Span<nint> destination);
+        ref readonly T GetPinnableReference();
+        void GetStrides(System.Span<nint> destination);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NIndex> startIndex);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NRange> range);
+        TSelf Slice(scoped System.ReadOnlySpan<nint> start);
+        bool TryCopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        bool TryFlattenTo(System.Span<T> destination);
+    }
+    public partial interface ITensor<TSelf, T> : System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable, System.Numerics.Tensors.IReadOnlyTensor<TSelf, T> where TSelf : System.Numerics.Tensors.ITensor<TSelf, T>
+    {
+        bool IsReadOnly { get; }
+        new T this[scoped System.ReadOnlySpan<System.Buffers.NIndex> indexes] { get; set; }
+        new TSelf this[scoped System.ReadOnlySpan<System.Buffers.NRange> ranges] { get; set; }
+        new T this[scoped System.ReadOnlySpan<nint> indexes] { get; set; }
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan();
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NIndex> startIndex);
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> range);
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<nint> start);
+        void Clear();
+        static abstract TSelf Create(System.ReadOnlySpan<nint> lengths, bool pinned = false);
+        static abstract TSelf Create(System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides, bool pinned = false);
+        static abstract TSelf CreateUninitialized(System.ReadOnlySpan<nint> lengths, bool pinned = false);
+        static abstract TSelf CreateUninitialized(System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides, bool pinned = false);
+        void Fill(T value);
+        new ref T GetPinnableReference();
+    }
+    public readonly ref partial struct ReadOnlyTensorSpan<T>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public ReadOnlyTensorSpan(System.Span<T> span) { throw null; }
+        public ReadOnlyTensorSpan(System.Span<T> span, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
```

Comment:
This should be taking `System.ReadOnlySpan`, we're not mutating it and so taking ROSpan is more flexible.

Reply:
Fixed.